geodirectory-widgets/geodirectory_bestof_widget.php

```diff
@@ -0,0 +1,507 @@
+            </label>
+        </p>
+        <?php
+    }
+}
+register_widget('geodir_bestof_widget');
+
+
+function geodir_bestof_places_by_term($query_args) {
+
+    $widget_listings = geodir_get_widget_listings( $query_args );
+
+    $character_count = $query_args['excerpt_length'];
+
+    if ( !isset( $character_count ) ) {
+        $character_count = $character_count == '' ? 50 : apply_filters( 'bestof_widget_character_count', $character_count );
+    }
+    $template = apply_filters( "geodir_template_part-widget-listing-listview", geodir_plugin_path() . '/geodirectory-templates/widget-listing-listview.php' );
+
+
+    global $post, $map_jason, $map_canvas_arr;
+    $current_post = $post;
+    $current_map_jason = $map_jason;
+    $current_map_canvas_arr = $map_canvas_arr;
+    $geodir_is_widget_listing = true;
+
+    include( $template );
+
+    $geodir_is_widget_listing = false;
+
+    $GLOBALS['post'] = $current_post;
+    setup_postdata( $current_post );
+    $map_jason = $current_map_jason;
+    $map_canvas_arr = $current_map_canvas_arr;
+
+}
+
+//Ajax functions
+add_action( 'wp_ajax_geodir_bestof', 'geodir_bestof_callback' );
+add_action( 'wp_ajax_nopriv_geodir_bestof', 'geodir_bestof_callback' );
+function geodir_bestof_callback() {
+    check_ajax_referer( 'geodir-bestof-nonce', 'geodir_bestof_nonce' );
+    //set variables
+    $post_type = strip_tags(esc_sql($_POST['post_type']));
+    $post_limit = strip_tags(esc_sql($_POST['post_limit']));
+    $character_count = strip_tags(esc_sql($_POST['char_count']));
+    $taxonomy = strip_tags(esc_sql($_POST['taxonomy']));
+    $add_location_filter = strip_tags(esc_sql($_POST['add_location_filter']));
+    $term_id = strip_tags(esc_sql($_POST['term_id']));
+
+    $query_args = array(
+        'posts_per_page' => $post_limit,
+        'is_geodir_loop' => true,
+        'post_type' => $post_type,
+        'gd_location' => $add_location_filter ? true : false,
+        'order_by' => 'high_rating'
+    );
+
+    if ( $character_count ) {
+        $query_args['excerpt_length'] = $character_count;
+    }
+
+    $tax_query = array(
+        'taxonomy' => $taxonomy,
+        'field' => 'id',
+        'terms' => $term_id
+    );
+
+    $query_args['tax_query'] = array( $tax_query );
+    if ($term_id && $taxonomy) {
+        $term = get_term_by('id', $term_id, $taxonomy);
+        echo '<h3 class="bestof-cat-title">Best of '.$term->name.'<a href="' . add_query_arg( array('sort_by' => 'overall_rating_desc'), get_term_link( $term ) ) . '">'.__( "View all", GEODIRECTORY_TEXTDOMAIN ).'</a></h3>';
+    }
+    geodir_bestof_places_by_term($query_args);
+    wp_die();
+}
+
+//Javascript
+add_action( 'wp_footer', 'geodir_bestof_js' );
+function geodir_bestof_js() { ?>
+    <script type="text/javascript" >
+    jQuery(document).ready(function($) {
```

Comment:
#1 move JS to file, probably geodirectory.js is best
#2 use jQuery not $
#3 for ajax url  u will find these declared as global values geodir_ajax_url, might need to add a global nonce value.